test/module/irohad/torii/query_service_test.cpp

```diff
@@ -164,3 +176,121 @@ TEST_F(QueryServiceTest, InvalidWhenDuplicateHash) {
+  std::shared_ptr<shared_model::proto::QueryResponse> model_response;
+  std::shared_ptr<shared_model::interface::BlockQueryResponse> block_response;
+  iroha::protocol::Block block;
+
+  shared_model::crypto::Keypair keypair =
+      shared_model::crypto::DefaultCryptoAlgorithmType::generateKeypair();
+
+  const std::string ip = "127.0.0.1";
+  int port;
+};
+
+/**
+ * @given valid blocks query
+ * @when blocks query is executed
+ * @then valid blocks response is received and contains block emitted by query
+ * processor
+ */
+TEST_F(BlocksQueryServiceTest, FetchBlocksWhenValidQuery) {
+  auto blocks_query = std::make_shared<shared_model::proto::BlocksQuery>(
+      shared_model::proto::BlocksQueryBuilder()
+          .creatorAccountId("user@domain")
+          .createdTime(iroha::time::now())
+          .queryCounter(1)
+          .build()
+          .signAndAddSignature(
+              shared_model::crypto::DefaultCryptoAlgorithmType::
+                  generateKeypair())
+          .finish());
+
+  EXPECT_CALL(
+      *query_processor,
+      blocksQueryHandle(
+          Truly([&blocks_query](
+                    const std::shared_ptr<shared_model::interface::BlocksQuery>
+                        query) { return *query == *blocks_query; })))
+      .WillOnce(Return(rxcpp::observable<>::just(block_response)));
+
+  auto client = torii_utils::QuerySyncClient(ip, port);
+  std::vector<iroha::protocol::BlockQueryResponse> responses;
+  auto proto_blocks_query =
+      std::static_pointer_cast<shared_model::proto::BlocksQuery>(blocks_query);
+  client.FetchCommits(proto_blocks_query->getTransport(), responses);
+
+  ASSERT_EQ(responses.size(), 1);
+  auto response = responses.at(0);
+  ASSERT_TRUE(response.has_block_response());
+  ASSERT_EQ(response.block_response().block().SerializeAsString(),
+            block.SerializeAsString());
+}
+
+/**
+ * @given stateless invalid blocks query
+ * @when blocks query is executed
+ * @then block error response is received
+ */
+TEST_F(BlocksQueryServiceTest, FetchBlocksWhenInvalidQuery) {
+  EXPECT_CALL(*query_processor, blocksQueryHandle(_)).Times(0);
+
+  auto blocks_query = std::make_shared<shared_model::proto::BlocksQuery>(
+      TestUnsignedBlocksQueryBuilder()
+          .creatorAccountId("asd@@domain")  // invalid account id name
+          .createdTime(iroha::time::now())
+          .queryCounter(1)
+          .build()
+          .signAndAddSignature(
+              shared_model::crypto::DefaultCryptoAlgorithmType::
+                  generateKeypair())
+          .finish());
+
+  auto client = torii_utils::QuerySyncClient(ip, port);
+  std::vector<iroha::protocol::BlockQueryResponse> responses;
+  auto proto_blocks_query =
+      std::static_pointer_cast<shared_model::proto::BlocksQuery>(blocks_query);
+  client.FetchCommits(proto_blocks_query->getTransport(), responses);
+
+  ASSERT_EQ(responses.size(), 1);
+  auto response = responses.at(0);
+  ASSERT_TRUE(response.has_error_response());
```

Comment:
Why don't we check value of the error?

Reply:
You right. Now block error response is very dumb. We should update it in future.